pkg/crypto/rings/client.go

```diff
@@ -194,3 +209,62 @@ func (rc *ringClient) addressesToPubKeys(
 	}
 	return pubKeys, nil
 }
+
+// getDelegationsAtBlock returns the appropriate delegations for the given
+// application and target block height.
+// The delegations selection is determined by the following:
```

Comment:
I find this a bit confusing. Here is how I thought it would work:

Why is it not just this?

```go
app.ActiveDelgations = list[addr]
app.ArchivedDelegations = map[session_end_height_where_it_invalid][list[addr]]

sessionEndHeight = getSessionEndHeight(blockHeight)

return app.ActiveDelgations + filter(app.ArchivedDelegations, key <= sessionEndHeight)
```